From 2c216c83f449225a99fd2cb08bb054306c153918 Mon Sep 17 00:00:00 2001
From: Charles Stevens <charles.stevens@compasses.com>
Date: Wed, 28 Jul 2021 16:49:12 -0500
Subject: [PATCH 54/57] RSOM-102 - Add SPI controller nodes and SPIDEV for SPI2
 and SPI3

arm64: dts: Beacon RZ/G2 baseboard add SPIDEV nodes for testing by default.

Signed-off-by: Charles Stevens <charles.stevens@compasses.com>
---
 .../dts/renesas/beacon-renesom-baseboard.dtsi | 40 ++++++++++++++++++-
 1 file changed, 38 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi b/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
index 109ac85e923e..696bdd7f3529 100644
--- a/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
+++ b/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
@@ -611,6 +611,16 @@
 		function = "msiof1";
 	};
 
+	msiof2_pins: msiof2 {
+		groups = "msiof2_clk_a", "msiof2_rxd_a", "msiof2_txd_a";
+		function = "msiof2";
+	};
+
+	msiof3_pins: msiof3 {
+		groups = "msiof3_clk_c", "msiof3_rxd_c", "msiof3_txd_c";
+		function = "msiof3";
+	};
+
 	pwm0_pins: pwm0 {
 		groups = "pwm0";
 		function = "pwm0";
@@ -729,7 +739,33 @@
 
 	spidev1@0 {
 	       compatible = "spidev";
-	       spi-max-frequency = <5000000>;
+	       spi-max-frequency = <8000000>;
+	       reg = <0>;
+	};
+};
+
+&msiof2 {
+	pinctrl-0 = <&msiof2_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+	cs-gpios = <&gpio1 8 GPIO_ACTIVE_LOW>;
+
+	spidev1@0 {
+	       compatible = "spidev";
+	       spi-max-frequency = <8000000>;
+	       reg = <0>;
+	};
+};
+
+&msiof3 {
+	pinctrl-0 = <&msiof3_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+	cs-gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
+
+	spidev1@0 {
+	       compatible = "spidev";
+	       spi-max-frequency = <8000000>;
 	       reg = <0>;
 	};
 };
@@ -864,7 +900,7 @@
 };
 
 &msiof_ref_clk {
-	clock-frequency = <50000000>;
+	clock-frequency = <134000000>;
 };
 
 &ehci0 {
-- 
2.17.1

